snuba/replacer.py

```diff
@@ -93,7 +93,7 @@ def __init__(self, clickhouse, dataset, metrics=None):
         self.dist_table_name = dataset.get_schema().get_table_name()
         self.metrics = metrics
         self.__all_column_names = [col.escaped for col in dataset.get_schema().get_columns()]
-        self.__required_columns = [col.escaped for col in dataset.get_required_columns()]
+        self.__required_columns = [col.escaped for col in dataset._get_required_columns()]
```

Comment:
If you need these methods from outside the events dataset class itself, they should not be preceded by the underscore by convention. Is there a specific reason for adding the underscore?
Other followup on this: the replacer is very tightly coupled with events in the current design. Since you are moving these methods out of the dataset parent class, I would suggest you go ahead and ensure the replacer script will only load the events dataset.

Reply:
👍 Renamed methods that are used from outside `events`. I guess [this](https://github.com/getsentry/snuba/blob/master/snuba/cli/replacer.py#L19) will prevent any other dataset option to be passed to the replacer